### ISSUE TITLE
Support formal-informal language variations

### ISSUE THREAD

**Issue by markusgeert** [consul/consul#5125]
### Why is this Needed?

[comment]: # (Describe the problem and why this task is needed. Provide a description of the current state, what you would like to happen, and what actually happened)

In a lot of languages (german, french, dutch, etc.) there exists a difference between [formal and informal language](https://en.wikipedia.org/wiki/T%E2%80%93V_distinction). In the Dutch language translations there is no consistency as to which mode is used ([informal](https://crowdin.com/translate/consul/411/en-nl?filter=basic&value=0#58716), [formal](https://crowdin.com/translate/consul/406/en-nl?filter=basic&value=3#52163)). 

Would it be possible to support two sets of language files for each language that uses different words in formal and informal speak?

If this is too much work and/or not a priority, it might be a good idea to pick one mode (formal or informal) per language, to at least enforce consistent communication? Let met know what you think.

**Comment by javierm** [comment]
Hi, @markusgeert :smile:.

> If this is too much work and/or not a priority, it might be a good idea to pick one mode (formal or informal) per language, to at least enforce consistent communication?

You're right; we should always use one mode. In Spain's Spanish, for instance, we try to use the informal mode because in Spain this mode is very common when talking to the general population.

In the case of Dutch, perhaps the institutions using Consul in the Netherlands could agree on which mode should be used and maybe get in touch with translators via Crowdin :thinking:. Other suggestions are welcome!

Regarding support for both modes, it's something we've been thinking about but so far we haven't found a maintainable way to do so.

**Comment by markusgeert** [comment]
Yep, I agree that it's quite a hard problem. In the netherlands both modes are used in different organizations, depending on location, target audience, message etc. So there isn't a general consensus on what to use unfortunately.

I'm not really familiar with Crowdin, but based on your response I assume they don't provide a way to add alternative translation in the same way singular and plural modes are handled?

It also seems more of a feature request for Crowdin than for Consul.

**Comment by markusgeert** [comment]
> Hi @markusgeert and @javierm, 
> 
> 
> 
> Reading the Crowdin docs, it seems we can add a [custom target language](https://crowdin.com/blog/2014/11/28/custom-languages-new-wordcount-algorithm) with a custom locale code in Crowdin and use it in CONSUL so we can have both Dutch versions co-existing in the application. The same applies to every other language.

This seems like a viable solution. I'm not really clear on this from the docs, but is it then also possible to fall back on the original language, so that for example the formal language only has to contain the strings for which it applies and use the 'default' strings otherwise? Otherwise this requires maintaining two very similar language files.

**Comment by Senen** [comment]
Hi @markusgeert 

> Another feature where this could be useful might be in having different names for the same concept. For example in Dutch, we sometimes want to call the Debates section 'Debatten' (more political, formal) or 'Discussies' (more informal, discussions). Currently we can use custom language files for this which works okay, but it is not ideal in terms of transferability and maintainability.

I think the situation you describe is very common in most of the languages. For example, we installed CONSUL for many institutions from Spain, mostly government-related institutions, and almost all of them want to reword something.

There is a feature in CONSUL called [Custom Information Texts](https://demo.consulproject.org/admin/site_customization/information_texts) that allow administrators to change the translation from most of the I18n keys. This helps a lot institutions to rename some words like "Debates" to "Discusiones" or whatever the institution wants.

**Comment by lucialuzuriaga** [comment]
Closing due to inactivity. As mentioned, Consul has the option to add [custom translation texts](https://docs.consuldemocracy.org/tech_docs/customization/translations). Additionally, since 2026 it also has AI translations for dynamic content, and we’re actively working in further improving translations in the platform.
